app/src/main/java/com/vitorpamplona/amethyst/ui/components/RobohashAsyncImage.kt

```diff
@@ -56,37 +52,22 @@ fun RobohashFallbackAsyncImage(
     colorFilter: ColorFilter? = null,
     filterQuality: FilterQuality = DrawScope.DefaultFilterQuality
 ) {
-    var loading by remember { mutableStateOf(false) }
-    var error by remember { mutableStateOf(false) }
-
-    Box {
-        AsyncImage(
-            model = model,
-            contentDescription = contentDescription,
-            modifier = modifier,
-            onLoading = { loading = true },
-            onSuccess = { loading = false; error = false },
-            onError = { error = true },
-            alignment = alignment,
-            contentScale = contentScale,
-            alpha = alpha,
-            colorFilter = colorFilter,
-            filterQuality = filterQuality
-        )
+    val context = LocalContext.current
+    val painter = rememberAsyncImagePainter(model = Robohash.imageRequest(context, robot))
 
-        if (loading || error) {
-            RobohashAsyncImage(
-                robot = robot,
-                contentDescription = contentDescription,
-                modifier = modifier,
-                alignment = alignment,
-                contentScale = contentScale,
-                alpha = alpha,
-                colorFilter = colorFilter,
-                filterQuality = filterQuality
-            )
-        }
-    }
+    AsyncImage(
+        model = model,
+        contentDescription = contentDescription,
+        modifier = modifier,
+        placeholder = painter,
+        fallback = painter,
+        error = painter,
```

Comment:
Apparently we can just do this to use a fallback and remove the makeshift Box. It seems to load faster for me and no noticeable blink, can you check it out again @vitorpamplona?